cms/djangoapps/contentstore/views/item.py

```diff
@@ -554,6 +585,18 @@ def _copy_to_other_course(user, xblock, course_dst_id):
                     broken_blocks=broken_blocks)
     return {'id': str(xblock.location), 'broken_blocks': broken_blocks}
 
+def _copy_unit_to_library(user, xblocks, library_dst_id):
+    broken_blocks = {}
+    library_key = usage_key_with_run(library_dst_id)
+    course_key = library_key.course_key
+    processed_ids = []
+    for xblock in xblocks:
+        for xblock_child in xblock.get_children():
+            processed_ids.append(xblock_child.location)
+            _duplicate_item(library_key, xblock_child.location, user=user, display_name=xblock_child.display_name, course_key=course_key,
```

Comment:
We must check block type of the `xblock_child`. To the library we can copy only: `problem` or `html` or `video`